make_ocr_features/utils/OCR_word_selection.py

```diff
@@ -0,0 +1,289 @@
+    'ingredient', 'ingredients', 'kcal', 'kg', 'kj', 'kl', 'l', 'la', 'le',
+    'mg', 'ml', 'net', 'net', 'of', 'or', 'per', 'poids', 'pour', 'total',
+    'valeur', 'which', 'wt', 'www'
+    ]
+
+### TFIDF TRICK
+
+def get_best_tfidf_words_list(sub_df_idx:int, feature_names:np.ndarray, tfidf_matrix,  n_words_to_pick:int=120)->str:
+    """ extracts a list with N highest tf-idf score, from a document.
+    Parameters
+    ----------
+    sub_df_idx: int
+        index of the document in the sub-dataframe (filtered dataframe on a given language).
+    feature_names:np.ndarray
+        name of the features given by vectorizer.get_feature_names_out()
+    tfidf_matrix: scipy.sparse.csr.csr_matrix
+        tfidf sparse matrix
+    N_words_to_pick: int, default = 120
+        total words to pick.
+
+    Returns
+    -------
+    words_picked: list:
+        list of N words
+    """
+    indexes_sorted_scores = np.argsort(tfidf_matrix[sub_df_idx].data)[-n_words_to_pick:]
+    indices_to_pick = tfidf_matrix[sub_df_idx].indices[indexes_sorted_scores]
+    words_picked = feature_names[indices_to_pick]
+    return words_picked
+
+
+def get_best_tfidf_words_cleaned(sub_df_idx:int, feature_names:np.ndarray, text_series:pd.Series, tfidf_matrix, freq_dict: dict,  n_max_words_to_pick:int=120)->str:
+    """get best tfidf unique words and keeps the original order and the original casing.
+    if a word has a frequence <=2 in all documents, it is filtered.
+
+    Parameters
+    ----------
+    sub_df_idx: int
+        index of the document in the sub-dataframe (filtered dataframe on a given language)
+    feature_names:np.ndarray
+        name of the features given by vectorizer.get_feature_names_out()
+    text_series: pd.Series
+        pd.Series containing OCR text
+    tfidf_matrix: scipy.sparse.csr.csr_matrix
+        tfidf sparse matrix.
+    freq_dict: dict
+        dictionary of word frequencies in the whole dataset.
+    n_max_words_to_pick: int, default = 120
+        total words to pick at most, it may be less than 120 because duplicated words and infrequent words are removed 
+
+    Returns
+    -------
+    words_picked: list:
+        list of N words
+    """
+    word_list = get_best_tfidf_words_list(sub_df_idx, feature_names, tfidf_matrix, n_max_words_to_pick)
+    df_text = text_series.iloc[sub_df_idx]
+    #this step is made to get the words original order.
+    if freq_dict is None:
+        words_picked = " ".join([str(w) for w in df_text.split() if str(w).lower() in word_list])
+    else:
+        words_picked = " ".join([str(w) for w in df_text.split() if (str(w).lower() in word_list and freq_dict[str(w).lower()]>2)])
+    return remove_duplicates(words_picked)
+
+def get_freq_dict(text_series):
+    """ computes word frequencies in text_series"""
+    freq_dict = defaultdict(int)
+    for text in text_series:
+        for word in text.split():
+            w_lower = str(word).lower()
```

Comment:
Nitpick: I mentioned it in another comment as well: Why the `str()` call here (and other places)? Aren't we guaranteed that `text.split()` returns strings?